React/Views/RCTModalHostView.m

```diff
@@ -89,6 +101,13 @@ - (void)insertReactSubview:(UIView *)subview atIndex:(NSInteger)atIndex
   RCTAssert(_reactSubview == nil, @"Modal view can only have one subview");
   [super insertReactSubview:subview atIndex:atIndex];
   [_touchHandler attachToView:subview];
+#if TARGET_OS_TV
+  if (_onRequestClose) {
```

Comment:
In my testing, `onRequestClose` gets set before `insertReactSubview`.  We definitely don't want the gesture recognizer to be added unless `onRequestClose` is implemented, otherwise the default menu button action won't happen.